src/Stratis.SmartContracts.Core.Validation/Primitives.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Stratis.SmartContracts.Core.Validation
+{
+    public static class Primitives
+    {
+        public static IEnumerable<Type> Types { get; } = new []
+        {
+            typeof(bool),
+            typeof(byte),
+            typeof(sbyte),
+            typeof(char),
+            typeof(int),
+            typeof(uint),
+            typeof(long),
+            typeof(ulong),
+            typeof(string),
+            typeof(byte[])
```

Comment:
Only partially related to the PR - not blocking, just a question: Do we need this line? I think it might be redundant with allowing Arrays?

Reply:
And do we have tests for arrays of disallowed types failing? Reminder to myself to check.